crypto/sleep.c

```diff
@@ -12,30 +12,37 @@
 
 /* system-specific variants defining OSSL_sleep() */
 #if defined(OPENSSL_SYS_UNIX) || defined(__DJGPP__)
-#include <unistd.h>
+# if defined(__TANDEM) && !defined(_REENTRANT)
 
+#  include <cextdecs.h(PROCESS_DELAY_)>
 void OSSL_sleep(uint64_t millis)
 {
-# ifdef OPENSSL_SYS_VXWORKS
-    struct timespec ts;
-
-    ts.tv_sec = (long int) (millis / 1000);
-    ts.tv_nsec = (long int) (millis % 1000) * 1000000ul;
-    nanosleep(&ts, NULL);
-# elif defined(__TANDEM) && !defined(_REENTRANT)
-#   include <cextdecs.h(PROCESS_DELAY_)>
-
     /* HPNS does not support usleep for non threaded apps */
     PROCESS_DELAY_(millis * 1000);
-# else
+}
+# elif defined(__DJGPP__) || defined(__TANDEM)
```

Comment:
Could you please add something like `|| defined(OPENSSL_USE_USLEEP)` ? This way there is a workaround for other potentially broken platforms.

Reply:
Ok

Reply:
Fixed.  I rearranged a little more for better logic